src/basic/sigbus.c

```diff
@@ -25,9 +25,17 @@ static unsigned n_installed = 0;
 static void* volatile sigbus_queue[SIGBUS_QUEUE_MAX];
 static volatile sig_atomic_t n_sigbus_queue = 0;
 
-static void sigbus_push(void *addr) {
+void sigbus_push(void *addr) {
+        unsigned long ul;
+        void *aligned;
+
         assert(addr);
 
+        ul = (unsigned long) addr;
+        ul = ul / page_size();
+        ul = ul * page_size();
+        aligned = (void*) ul;
```

Comment:
When you touch the code, then please use `PAGE_ALIGN()` defined in memory-util.h.